src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,172 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    private static class Node<T> {
+        private T element;
+        private Node<T> next;
+        private Node<T> prev;
+
+        private Node(Node<T> prev, T element, Node<T> next) {
+            this.element = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
     @Override
     public void add(T value) {
+        addLast(value);
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexToAdd(index);
+        if (index == size) {
+            addLast(value);
+        } else {
+            addBefore(value, getNodeByIndex(index));
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list != null) {
```

Comment:
do we need this check for null?

Reply:
Without null check I can catch NPE. It's not about tests. I don't write code just to pass the tests. It's about protection.